### DISCUSSION TITLE
Go to Definition shows up multiple files

### DISCUSSION THREAD

**Discussion by adam3141** [microsoft/vscode-cpptools#9396] [Q&A] [answered]
Sorry if this isn't the correct place to ask this, and I am not sure if it is a bug or something else so I was hoping someone from the community might have some insight - or whether it is actually buggy behaviour (it seems like it to me)

I am writing firmware for a work project and we need to support a couple of different microcontrollers. Now I have tried to set the configuration as follows
```
"configurations": [
        {
            "name": "Linux - Nuvoton 126 EVK",
            "includePath": [
                "${workspaceFolder}/nulib/nuc126/CMSIS/Include",
                "${workspaceFolder}/nulib/nuc126/Device/Nuvoton/NUC126/Include",
                "${workspaceFolder}/nulib/nuc126/StdDriver/inc",
                "${workspaceFolder}/nulib"
            ],
```
I also need to support a nuc1262 and so for a different configuration (I actually haven't added it yet) those nuc126 directories would be nuc1262.

Now, I have tried to be as specific as possible, avoiding any globbing behaviour because the same header files exist for both microcontrollers.

I have the **Linux - Nuvoton 126 EVK** selected as my configuration and when I try to Ctrl+click on a symbol it brings up the following
![image](https://user-images.githubusercontent.com/5292494/171810667-c0bea132-61a8-4583-81bf-4b6f649cfb82.png)

and it clearly shows that it is also searching in the path nulib/nuc1262 and according to my includePath list, it shouldn't be searching as I don't have ${workspaceFolder}/nulib/** 

For my situation, it shows the correct gpio.h file and I am wondering if this is because the nuc126 comes before nuc1262. My concern is that if I switch to the **Linux - Nuvoton 1262 EVK** configuration when I eventually add it, the gpio.h it selects will still be from nuc126 and I would manually have to select the correct one each time.

I guess my questions are:
- Is this expected behaviour?
- Am I setting something incorrectly?
- Is this an actual bug? How is it seeing the nuc1262 directory without the glob pattern specified?

Thanks

**Comment by sean-mcmanus** [comment, marked as answer]
You need to use something like
```
"browse":
{
"path": [
                "${workspaceFolder}/nulib/nuc126/CMSIS/Include",
                "${workspaceFolder}/nulib/nuc126/Device/Nuvoton/NUC126/Include",
                "${workspaceFolder}/nulib/nuc126/StdDriver/inc"
]
}
```
i.e. the `"${workspaceFolder}/nulib"` from the includePath needs to be omitted since it causes that folder to be recursivey parsed for workspace symbols.

**Reply by adam3141** [reply]
Thanks for that - I will certainly give that a shot, however I am curious why it should be recursively parsed. I thought it would only be recursively parsed if I did this `${workspaceFolder}/nulib/**`

Edit: Thanks Sean. That seemed to work.

**Reply by sean-mcmanus** [reply]
The design is somewhat confusing -- the "includePath" property is not recursive by default (requires the "**") however that is by default copied to the "browse.path" property, which is recursive by default (unless "*" is added to the end).
